src/org/exist/collections/Collection.java

```diff
@@ -2305,4 +2303,32 @@ public QNameRangeIndexSpec getIndexByQNameConfiguration(final DBBroker broker, f
         final IndexSpec idxSpec = getIndexConfiguration(broker);
         return (idxSpec == null) ? null : idxSpec.getIndexByQName(qname);
     }
+
+    @Override
+    public ResourceMetadata getMetadata() {
+        return new CollectionMetadata();
+    }
+
+    class CollectionMetadata implements ResourceMetadata {
+
+        @Override
```

Comment:
it's serializing to backup (for example). `Optional` adding complexity here, IMHO